### PR TITLE
Add KUBERNETES_CONTAINER_NAME env var

### PR DESCRIPTION
**Pull request by jandubois**
This is needed by configgin to export properties only from the container whose name matches the role name (and not the loggregator agent sidecar). There seems to be no mechanism to reference the `container.name` field in the env section, so the value has to be provided by fissile.

[jsc#CAP-761]

### CODE REVIEW

File: model/mustache.go
Comment (mook-as) [inline model/mustache.go:88]: So the container name is the name of the instance group, not the container? Oh, because the "instance group" here does _not_ include the addons / colocated containers.

We should probably include a comment here so I don't get confused again in the future…

**Review by mook-as — CHANGES_REQUESTED** [review]
Needs a comment.

Do we have any tests that would show what value we get for `KUBERNETES_CONTAINER_NAME` for colocated containers? We should probably get one of those… But maybe I just missed the relevant change.

File: model/mustache.go
Comment (jandubois) [reply model/mustache.go:88]: The co-located container has its own instance group; it is not part of the main instance group:

https://github.com/SUSE/scf/blob/develop/container-host-files/etc/scf/config/role-manifest.yml#L2590-L2591

What do you want the comment to say?

Comment (mook-as) [reply model/mustache.go:88]: Something to indicate that the `instance group` here _isn't_ the pod name because…

**Review by mook-as — APPROVED** [review]
:+1: waiting on CI before merge

**Pull request merged**
